test/conn_test.go

```diff
@@ -1470,90 +1470,183 @@ func TestDefaultOptionsDialer(t *testing.T) {
 	}
 }
 
-func TestCustomFlusherTimeout(t *testing.T) {
-	t.Skip("broken test")
+type lowWriteBufferDialer struct{}
+
+func (d *lowWriteBufferDialer) Dial(network, address string) (net.Conn, error) {
+	c, err := net.Dial(network, address)
+	if err != nil {
+		return nil, err
+	}
+	c.(*net.TCPConn).SetWriteBuffer(100)
+	return c, nil
+}
 
+func TestCustomFlusherTimeout(t *testing.T) {
 	s := RunDefaultServer()
 	defer s.Shutdown()
 
-	opts := &nats.Options{
-		Servers: []string{nats.DefaultURL},
-
-		// Reasonably large flusher timeout will not induce errors
-		// when we can flush fast
-		FlusherTimeout: 10 * time.Second,
-	}
-	nc1, err := opts.Connect()
+	// Reasonably large flusher timeout will not induce errors
+	// when we can flush fast
+	nc1, err := nats.Connect(nats.DefaultURL, nats.FlusherTimeout(10*time.Second))
 	if err != nil {
 		t.Fatalf("Expected to be able to connect, got: %s", err)
 	}
-	doneCh := make(chan struct{})
-	payload := ""
-	for i := 0; i < 8192; i++ {
-		payload += "A"
-	}
-	payloadBytes := []byte(payload)
+	doneCh := make(chan struct{}, 1)
+	// We want to have a payload size that is big enough so that after
+	// few publish, the socket buffer will be full and produce the timeout.
+	// Since we try to produce the error in the flusher and not the publish
+	// call itslef, use a size that is a bit less than the internal
```

Comment:
```suggestion
	// call itself, use a size that is a bit less than the internal
```

Reply:
Thanks, I fixed and push-force the changes.